internal/authentication/authentication.go

```diff
@@ -0,0 +1,63 @@
+package authentication
+
+import (
+	"errors"
+	"log/slog"
+	"net/http"
+	"strings"
+
+	"github.com/gpuctl/gpuctl/internal/femto"
+	"github.com/gpuctl/gpuctl/internal/types"
+)
+
+var (
+	NotAuthenticatedError   = errors.New("User does not have a valid authentication token")
+	InvalidCredientalsError = errors.New("Invalid credientals for creating an authentication token")
+)
+
+const (
+	AuthCookieField = "auth_cookie"
+)
+
+type AuthToken = string
```

Comment:
Then lets call it `BearerToken`, because that's how it's used. Some of the tokens being used in authorization not being authorization tokens is just going to make a mess.

Reply:
Not quite. Other tokens, like CSRF and session tokens **are** authorisation tokens. It may be worth renaming this one, but all tokens (which are either cookies, bearers or tagged along with all responses) are authorisation tokens given they are tokens which actually necessarily do authorise they user's action.